### PR TITLE
Add cloud provider migration

### PR DESCRIPTION
**Pull request by galal-hussein**
- https://github.com/rancher/migration-agent/issues/4
- https://github.com/rancher/migration-agent/issues/5

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -71,7 +90,8 @@ func ExportClusterConfiguration(ctx context.Context, fullState *cluster.FullStat
 	return ioutil.WriteFile(configPath, data, 0644)
 }
 
-func getClusterConfig(fullState *cluster.FullState, nodeName string) map[string]string {
+// constructing an rke2 config file for server options
```
Comment (briandowns) [inline pkg/config/config.go:93]: Can you update this to be in doccomment format?

**Review by briandowns — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

File: main.go [hunk 2]
```diff
@@ -99,15 +109,31 @@ func main() {
...
 		logrus.Fatal(err)
 	}
 }
 
 func run(c *cli.Context) {
+	// set up logging to disk
+	if err := os.MkdirAll(filepath.Dir(logFile), 0755); err != nil {
+		logrus.Fatalf("failed to create logging directory: %v", err)
+	}
+	f, err := os.OpenFile(logFile, os.O_WRONLY|os.O_CREATE, 0755)
```
Comment (dweomer) [inline main.go:130]: Anyone can read this file? Migration logs tend to expose unexpected things over time, maybe we should lock this down more?

Comment (dweomer) [inline main.go:130]:
```suggestion
	f, err := os.OpenFile(logFile, os.O_WRONLY|os.O_CREATE, 0600)
```

Comment (galal-hussein) [reply main.go:130]: will fix

**Review by dweomer — APPROVED** [review]
LGTM (with perms caveat)

**Pull request merged**